Lib/test/test_ctypes/test_arrays.py

```diff
@@ -267,6 +267,25 @@ def test_bpo36504_signed_int_overflow(self):
     def test_large_array(self, size):
         c_char * size
 
+    @threading_helper.requires_working_threading()
+    @unittest.skipUnless(Py_GIL_DISABLED, "only meaningful if the GIL is disabled")
+    def test_thread_safety(self):
```

Comment:
For me, with a `t` build, the test passes even without the rest of the PR's changes.
How can I make it fail?